test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,350 @@
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return nil
+	}
+
+	assert.Eventually(t, func() bool {
+		t.Logf("trying to curl the fqdn %v", testFullyQualifiedDomainName)
+		err = curlFqdn(toolBoxPodName, toolboxContainerName, testFullyQualifiedDomainName)
+		return assert.NoError(t, err)
+	}, 2*time.Second, 100*time.Millisecond)
+
+	// DIG to get actual IP , to be sure.
+	digResponse, err := k8sUtils.runDNSQuery(toolBoxPodName, toolboxContainerName, data.testNamespace, testFullyQualifiedDomainName, false, customDnsService.Spec.ClusterIP)
+	require.NoError(t, err, "failed to get IP of FQDN using DIG from toolbox pod : %v", err)
+	fqdnIp := digResponse.String()
+
+	t.Logf("received ip using dig for test fqdn %+v ", fqdnIp)
+
+	ipv4, ipv6 = extractIPs(agnHostPodIps[1])
+	if ipv6 != "" {
+		t.Logf("New IPs to update to DNS | ipv4 : %v, ipv6 : %+v", ipv4, ipv6)
+	} else {
+		t.Logf("New IPs to update to DNS | ipv4 : %v", ipv4)
+	}
+
+	// Update the custom DNS configMap
+	UpdatedCustomDNSconfig := createDnsConfig(t, ipv4, ipv6)
+
+	customDnsConfigMapObject.Data = UpdatedCustomDNSconfig
+	err = data.UpdateConfigMap(customDnsConfigMapObject)
+	require.NoError(t, err, "failed to update configmap with new IP : %v", err)
+	if ipv6 != "" {
+		t.Logf("successfully updated dns configMap with new IPs | ipv4 : %+v, ipv6 :%+v", ipv4, ipv6)
+	} else {
+		t.Logf("successfully updated dns configMap with new IPs | ipv4 : %+v", ipv4)
+	}
+
+	require.NoError(t, data.setPodAnnotationToRandomValue(data.testNamespace, customDnsPodName, randomPatchAnnotationKey), "failed to update custom dns pod annotation.")
+
+	defer setDnsServerAddressInAntrea(t, data, "")
+
+	assert.Eventually(t, func() bool {
+		t.Logf("trying to curl the existing cached IP of the domain  %v", fqdnIp)
+		err = curlFqdn(toolBoxPodName, toolboxContainerName, fqdnIp)
+		if err != nil {
+			t.Logf("The test failed because of error  %+v", err)
+		}
+		return assert.Error(t, err)
+	}, eventualWaitTime, 1*time.Second)
+
+}
+
+func setDnsServerAddressInAntrea(t *testing.T, data *TestData, dnsServiceIP string) {
+	agentChanges := func(config *agentconfig.AgentConfig) {
+		config.DNSServerOverride = dnsServiceIP
+	}
+	err := data.mutateAntreaConfigMap(nil, agentChanges, false, true)
+	require.NoError(t, err, "Error when setting up customDNS server IP in Antrea configmap : %v", err)
+
+	if dnsServiceIP == "" {
+		t.Logf("removing dns server IP from antrea agent as part of teardown")
+	} else {
+		t.Logf("dns server value set to %+v in antrea \n", dnsServiceIP)
+	}
+
+}
+
+func createDnsConfig(t *testing.T, ipv4Address, ipv6Address string) map[string]string {
+	configMapData, _ := generateCoreFile(t, ipv4Address, ipv6Address)
+	configData := map[string]string{
+		"Corefile": configMapData,
+	}
+
+	return configData
+}
+
+func generateCoreFile(t *testing.T, ipv4Address, ipv6Address string) (string, error) {
```

Comment:
could we "merge" these 2 functions? maybe by making `generateCoreFile` a closure in `createDnsConfig`